api/src/main/java/engineering/everest/lhotse/api/rest/controllers/AdminOrganizationsController.java

```diff
@@ -0,0 +1,84 @@
+import engineering.everest.lhotse.api.rest.converters.DtoConverter;
+import engineering.everest.lhotse.api.rest.requests.NewOrganizationRequest;
+import engineering.everest.lhotse.api.rest.responses.OrganizationResponse;
+import engineering.everest.lhotse.axon.common.domain.User;
+import engineering.everest.lhotse.organizations.services.OrganizationsReadService;
+import engineering.everest.lhotse.organizations.services.OrganizationsService;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.UUID;
+
+import static java.util.stream.Collectors.toList;
+import static org.springframework.http.HttpStatus.CREATED;
+import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@RestController
+@RequestMapping("/api/admin/organizations")
+@Api(consumes = APPLICATION_JSON_VALUE, tags = "Organizations")
+public class AdminOrganizationsController {
+
+    private final DtoConverter dtoConverter;
+    private final OrganizationsService organizationsService;
+    private final OrganizationsReadService organizationsReadService;
+
+    @Autowired
+    public AdminOrganizationsController(DtoConverter dtoConverter,
+                                        OrganizationsService organizationsService,
+                                        OrganizationsReadService organizationsReadService) {
+        this.dtoConverter = dtoConverter;
+        this.organizationsService = organizationsService;
+        this.organizationsReadService = organizationsReadService;
+    }
+
+    @GetMapping
+    @ResponseStatus(OK)
+    @ApiOperation(produces = APPLICATION_JSON_VALUE, value = "Retrieves details of all organizations")
+    @AdminOnly
+    public List<OrganizationResponse> getAllOrganizations() {
+        return organizationsReadService.getOrganizations().stream()
+                .map(dtoConverter::convert)
+                .collect(toList());
+    }
+
+    @PostMapping
+    @ResponseStatus(CREATED)
+    @ApiOperation("Register a new organization")
+    @AdminOnly
+    public UUID registerOrganization(User requestingUser, @RequestBody @Valid NewOrganizationRequest request) {
+        return organizationsService.createRegisteredOrganization(requestingUser.getId(), request.getOrganizationName(),
```

Comment:
The thinking behind allowing multiple registrations for the same email address was to have a contrived example to demonstrate how such a situation can be handled by the saga and the command validation as it would be a business decision as to how to handle multiple pending registrations. 

I put a comment in the saga attempting to explain this.

Uniqueness is protected to the same level as the current command validation process as the final step of creating a user can still be rejected. (Cleaning up of the saga based on deadline expiry hasn't been implemented).

Reply:
Ok I had some misunderstanding previously. This command creates organisation directly (as in directly into database), but does *not* create an user (unlike org registration). The email uniqueness is required for users, but not organizations, i.e. multiple organizations can share the same contact email. This makes sense.

However, this command has no validator, so it is also possible to create an organization with an email that is *not* associated with any user. I understand all these requirements are contrived. So they are somewhat arbitrary. But I'd prefer to have a validation to prevent this scenario from happening as it feels a bit strange.